File: Daily Programmer/Easy Challenges/0198 - Words with Enemies/Solution.py
def results(word_one, word_two):
    for c in word_one + word_two:
        if c in word_one and c in word_two:
            word_one, word_two = word_one.replace(c, "", 1), word_two.replace(c, "", 1)
    return "Valley Letters: %s,%s\n%s" % (word_one, word_two, "Tie" if len(word_one) == len(word_two) else "Left Wins" if len(word_one) > len(word_two) else "Right Wins")


def main():
    words = input("Enter two words (space-delimited): ").split();
    print(results(words[0], words[1]))


if __name__ == '__main__':
    main()

File: Daily Programmer/Easy Challenges/0198 - Words with Enemies/Solution.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Enter two words (space-delimited): ");
        String arrWords[] = reader.nextLine().split(" ");
        System.out.println(results(reduceWord(arrWords[0], arrWords[1]), reduceWord(arrWords[1], arrWords[0])));
    }

    public static String reduceWord(String wordOne, String wordTwo) {
        for (char c: wordTwo.toCharArray()) {
            wordOne = wordOne.replaceFirst(Character.toString(c), "");
        }
        return wordOne;
    }

    public static String results(String wordOne, String wordTwo) {
        return String.format("Valley Letters: %s,%s%n%s", wordOne, wordTwo, (wordOne.length() == wordTwo.length()) ? "Tie" : (wordOne.length() > wordTwo.length() ? "Left Wins" : "Right Wins"));
    }

}